li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -87,6 +87,15 @@ boolean isCommittedOffsetDefined() {
   private final DeserializeStrategy<V> _deserializeStrategy;
 
   private long recordsSkipped = 0;
+  /**
+   * Gives an idea about how far behind the safe offset of a partition is from the watermark. Hence, the
+   * offset-watermark span or a partition is computed as (watermark minus safeoffset).
+   * This metric is a sum of
+   * {@link com.linkedin.kafka.clients.largemessage.DeliveredMessageOffsetTracker#offsetWatermarkSpan} for all
+   * assigned partitions that have currently undelivered large messages in the consumer. It is computed on query of
+   * the metric each time and hence, stores the last known value for this metric.
+   */
+  private long _consumerOffsetWatermarkSpan = -1;
```

Comment:
Metrics are often accessed from different threads, even in the consumer, and so shouldn't this be volatile.

Reply:
Makes sense. Looking at the code, though, this variable doesn't need to exist since the value is computed each time in the `getConsumerOffsetWatermarkSpan` method.